x-pack/metricbeat/module/meraki/device_health/devices.go

```diff
@@ -173,10 +186,14 @@ func getDeviceLicenses(client *meraki.Client, organizationID string, devices map
 	val, res, err := client.Organizations.GetOrganizationLicenses(organizationID, &meraki.GetOrganizationLicensesQueryParams{})
 	if err != nil {
 		// Ignore 400 error for per-device licensing not supported
-		if res.StatusCode() == 400 && strings.Contains(string(res.Body()), "does not support per-device licensing") {
+		if res != nil && res.StatusCode() == 400 && strings.Contains(string(res.Body()), "does not support per-device licensing") {
```

Comment:
I was thinking about this, if we add this check we fix the panic but we will go back to the issue in the first place, the function will return and the metric collection will stop just like before.

Reply:
Added an extra check https://github.com/elastic/beats/pull/44193/commits/6b6f6f19b18f33a7221d6db6cf2c4e98c7fd8db6 for `err` message that simply returns without an error but we do not know if that would be the error contents that is returned and ideally we want to replicate this.